cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationService.java

```diff
@@ -388,9 +388,10 @@ private static boolean hasCauseAssignableFrom( @Nonnull final Throwable t, @Nonn
         return ExceptionUtils.getThrowableList(t).stream().map(Throwable::getClass).anyMatch(cls::isAssignableFrom);
     }
 
-    private boolean preLookupCheckSuccessful( final String destinationName )
+    private boolean preLookupCheckSuccessful( final String destinationName, final DestinationOptions options )
     {
-        return getAllDestinationProperties()
+        return new ArrayList<>(
+            Cache.getOrComputeAllDestinations(options, this::getAllDestinationsByRetrievalStrategy).get())
```

Comment:
(Comment/Discussion)

I feel like with the motivation of this PR, the following would be more explicit while higher-level(?)
```suggestion
        final DestinationServiceRetrievalStrategy retrievalStrategy =
            DestinationServiceOptionsAugmenter
                .getRetrievalStrategy(options)
                .getOrElse(DestinationServiceRetrievalStrategy.CURRENT_TENANT);

        return getAllDestinationProperties(retrievalStrategy)
```